substrate/frame/offences/benchmarking/src/inner.rs

```diff
@@ -184,17 +184,22 @@ fn make_offenders<T: Config>(
 }
 
 #[cfg(test)]
-fn assert_all_slashes_applied<T>(offender_count: usize)
-where
+fn assert_all_slashes_applied<T>(
+	offender_count: usize,
+	reporter: <T as frame_system::Config>::AccountId,
+) where
 	T: Config,
 	<T as frame_system::Config>::RuntimeEvent: TryInto<pallet_staking::Event<T>>,
 	<T as frame_system::Config>::RuntimeEvent: TryInto<pallet_balances::Event<T>>,
 	<T as frame_system::Config>::RuntimeEvent: TryInto<pallet_offences::Event>,
 	<T as frame_system::Config>::RuntimeEvent: TryInto<frame_system::Event<T>>,
 {
-	// make sure that all slashes have been applied and TotalIssuance adjusted(BurnedDebt).
-	// deposit to reporter + reporter account endowed.
-	assert_eq!(System::<T>::read_events_for_pallet::<pallet_balances::Event<T>>().len(), 3);
+	// make sure reporter got deposited.
+	assert!(System::<T>::read_events_for_pallet::<pallet_balances::Event<T>>()
```

Comment:
From now on, pledge to not approve any PR that does this "event counting" shenanigans :D